Add the packaging metadata to build the bittorrent-tracker snap

elopio opened the pull request:
This is a package for the secure installation of apps that works in most Linux distributions.
Landing it upstream will enable builds for adventurous users.

elopio commented:
You can find more information about snapcraft here: http://snapcraft.io/

To test it in an Ubuntu 16.04 machine:

$ sudo apt install git snapcraft
$ git clone https://github.com/elopio/bittorrent-tracker
$ cd bittorrent-tracker
$ git checkout snapcraft
$ snapcraft
$ sudo snap install *.snap --dangerous
$ bittorrent-tracker

If you have any questions or comments, please let me know.

popey commented:
Does someone have time to review this? It would be great to land bittorrent-tracker in the snap store, especially given the frequency of updates/releases to the project.

feross commented:
I'm happy to merge this and maintain a snap. We should add a post-publish hook so that when `npm publish` is run we update the version in `snapcraft/yml` and publish a new snap.

feross commented on snapcraft.yaml:
```diff
@@ -0,0 +1,18 @@
+grade: devel
+confinement: strict
+
+parts:
```
This "parts" section exports a re-usable part named "tracker", right? Should we name it "bittorrent-tracker" to be clearer?

elopio replied:
it is internal to your project, not really used by anything else. So we can actually put whatever name we want. I've updated it to bittorrent-tracker.

elopio commented:
Thanks for the review @feross!
I've pushed a new revision.

The post-publish hook sounds good. I don't really know much about npm, so I'll keep an eye here to learn about what you do.
An alternative I've been experimenting with is to use the git tags. Take a look for example at https://github.com/magne4000/quassel-webserver/pull/223/files
In there, we are using travis to push a new version to edge every time master changes, and if the change is a new tag, we sed the version first.

yciabaud commented:
You can update the revision in the file using `PACKAGE_VERSION=$(grep version package.json | head -1 | awk -F: '{ print $2 }' | sed 's/[\",]//g'
| tr -d '[[:space:]]') && sed -i "s/^version: .*$/version: $PACKAGE_VERSION/" snapcraft.yaml`

elopio commented:
@feross thanks to your comments and some from other upstreams, we added a new tiny feature. When you define `version: git`, it will take the tag as the version, if there is one. I've updated my branch and added this to the yaml.
